org.eclipse.swtchart.export/src/org/eclipse/swtchart/export/images/ImageSupplier.java

```diff
@@ -15,52 +15,76 @@
 import org.eclipse.swt.graphics.GC;
 import org.eclipse.swt.graphics.Image;
 import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.ImageDataProvider;
 import org.eclipse.swt.graphics.ImageLoader;
-import org.eclipse.swt.graphics.Rectangle;
-import org.eclipse.swt.graphics.Transform;
+import org.eclipse.swt.graphics.PaletteData;
+import org.eclipse.swt.graphics.Point;
 import org.eclipse.swtchart.extensions.core.BaseChart;
 
 public class ImageSupplier {
 
+	// Image data provider used locally to represent chart image
+	private class ChartImageDataProvider implements ImageDataProvider {
+
+		private ImageData imageData;
+
+		public ChartImageDataProvider(int width, int height) {
+
+			PaletteData palette = new PaletteData(0xFF, 0xFF00, 0xFF0000);
+			imageData = new ImageData(width, height, 32, palette);
+		}
+
+		@Override
+		public ImageData getImageData(int zoom) {
+
+			if(zoom != 100)
+				return null;
+			return imageData;
+		}
+	}
+
+	/**
+	 * Save image data representation to a file with specific format (BMP, PNG or JPG)
+	 * 
+	 * @param imageData
+	 *            the image data representation
+	 * @param fileName
+	 *            the file (relative or full path)
+	 * @param format
+	 *            SWT.IMAGE_BMP, SWT.IMAGE_PNG or SWT.IMAGE_JPEG
+	 */
 	public void saveImage(ImageData imageData, String fileName, int format) {
 
 		ImageLoader imageLoader = new ImageLoader();
 		imageLoader.data = new ImageData[]{imageData};
 		imageLoader.save(fileName, format);
 	}
 
+	/**
+	 * Return image data representation of provided chart
+	 * 
+	 * @param baseChart
+	 *            the chart from which is create image data representation
+	 * @return image data representation
+	 */
 	public ImageData getImageData(BaseChart baseChart) {
 
-		Image image = new Image(baseChart.getDisplay(), baseChart.getBounds());
-		GC gc = new GC(image);
-		/*
-		 * Under macOS, the image is flipped.
-		 */
-		if(isMacOS()) {
-			Transform transform = new Transform(gc.getDevice());
-			transform.rotate(-180);
-			Rectangle bounds = image.getBounds();
-			transform.translate(-bounds.width, -bounds.height);
-			gc.setTransform(transform);
-			baseChart.print(gc);
-			transform.dispose();
-		} else {
-			baseChart.print(gc);
-		}
-		//
-		gc.dispose();
+		// Force to redraw chart immediately to be sure that any
+		// previous Shell dialog won't be a part of the copied image
+		baseChart.redraw();
+		baseChart.update();
+		// Chart size
+		Point baseChartSize = baseChart.getSize();
+		// Create the image provider
+		ChartImageDataProvider chartImageDataProvider = new ChartImageDataProvider(baseChartSize.x, baseChartSize.y);
+		// Copy chart into the image
+		Image image = new Image(baseChart.getDisplay(), chartImageDataProvider);
+		GC gc = new GC(baseChart);
+		gc.copyArea(image, 0, 0);
```

Comment:
The gc and image part should be enclosed in try / finally so the gc/image gets disposed whatever happens in between

Reply:
Done